website/docs/sql/external/formats.md

```diff
@@ -338,3 +339,85 @@ Hyper's support for the Iceberg format is still in an early state and
 should be considered experimental. We recommend to not use it in
 production workloads yet.
 :::
+
+## Apache Arrow Format {#external-format-arrow}
+
+The `FORMAT => 'arrowfile'` or `FORMAT => 'arrowstream'` option, 
+a file extension of `.arrow` or `.arrows` enables reading the [Apache Arrow
+format](https://arrow.apache.org/), which is a binary columnar format.
+Thus, data stored in Arrow is usually smaller than in text
+format. As data is stored in columnar format, Hyper does not need to
+read the whole file if only a subset of columns is selected but can
+instead only read the parts of the file in which selected columns are
+stored. This will speed up queries selecting only few columns of a file
+having a lot of columns. Opposing to (Parquet)[#external-format-parquet],
+we do not support loading partial Arrow files from S3 - if reading a 
+subset of all columns from S3, Parquet is the better option.
+
+We support [IPC streaming format](https://arrow.apache.org/docs/format/Columnar.html#ipc-streaming-format)/`.arrows`
+with `FORMAT => 'arrowstream'` and [IPC file format](https://arrow.apache.org/docs/format/Columnar.html#ipc-file-format)
+/Feather V2/`.feather`/`arrow` file format with `FORMAT => 'arrowfile'`.
+
+The only options supported by Arrow are the 
+[common format options](#common-format-options). The `COLUMNS` option 
+is required because we do not support schema inference.
+We provide a 1-to-1 mapping (table below) from Arrow types to SQL types and expect the user
+to choose these SQL types exactly. We do not perform any implicit casts.
+The `COMPRESSION` option is not supported.
+
+Arrow Type | Hyper Type | Notes
+----|----|----
+`Bool` | `Bool`
+`Int` signed, 8 bit | `SmallInt` |
+`Int` signed, 16 bit | `SmallInt` |
+`Int` signed, 32 bit | `Integer` |
+`Int` signed, 64 bit | `BigInt` |
+`Int` unsigned, 8 bit | `SmallInt` |
+`Int` unsigned, 16 bit | `Integer` |
+`Int` unsigned, 32 bit | `BigInt` |
+`Int` unsigned, 64 bit | `Numeric(20,0)` | Not recommended - processing is slower than other Int types.
+`FloatingPoint` with any precision | `DOUBLE PRECISION` | Half floats are not supported.
+`Decimal(precision, scale)` | `NUMERIC(precision, scale)` | We do not support precision > 38. We recommend using precision <= 18 if possible.
+`Date` | `Date` |
+`Time` | `Time` | Hyper does not support nano second precision and truncates it to micro seconds.
+`Timestamp` without timezone | `TIMESTAMP` | Hyper does not support nano second precision.
+`Timestamp` with timezone | `TIMESTAMPTZ` | Hyper does not support nano second precision.
+`UTF8` | `Text`
+
+
+Hyper supports only version 1.3. However, some data types, encodings and
+compressions are not supported. The following Arrow features are not supported:
+
+- Interval
+
+- LargeUtf8 type
+
+- Struct, lists, union, map types
+
+- Binary, LargeBinary, FixedSizeBinary
+
+- Half float type
+
+- Run end or dictionary encoding
+
+- Arrow internal compression is not supported
```

Comment:
also: dictionaries

Reply:
I'm not sure what to do here. The upper line states that we do not support `dictionary encoding` (in my mind this includes the fact that we do not read Dictionary Batches). The second line refers to compression like Z4 on any batches. I clarified the second line. Did you expect something else?

Reply:
ah, I missed that we combined both "run end" and "dictionary" encoding into a single line